pkg/controller/controller_instance_test.go

```diff
@@ -1771,6 +1771,52 @@ func TestReconcileServiceInstanceDeleteFailedProvisionWithoutRequest(t *testing.
 	assertNumEvents(t, events, 0)
 }
 
+// TestReconcileServiceInstanceDeleteFailedProvisionWithoutRequestWithActiveBinding tests
+// that an instance that failed to provision without making a provision request
+// will be finalized even if it has binding(s) and no deprovision request will be sent to the broker.
+func TestReconcileServiceInstanceDeleteFailedProvisionWithoutRequestWithActiveBinding(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeClusterServiceBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.ClusterServiceBrokers().Informer().GetStore().Add(getTestClusterServiceBroker())
+	sharedInformers.ClusterServiceClasses().Informer().GetStore().Add(getTestClusterServiceClass())
+	sharedInformers.ClusterServicePlans().Informer().GetStore().Add(getTestClusterServicePlan())
+	sharedInformers.ServiceBindings().Informer().GetStore().Add(getTestServiceBinding())
```

Comment:
`getTestServiceBinding()` is a bit out of place here, because the resource it's creating is supposed to represent a binding that was created at the broker, and thus it has its `UnbindStatus` set to `Required` (looks like we don't set the Ready condition on it, though, which is a bit bizzare). Maybe we should create a new `getTestServiceBindingInactive()` function that represents a fresh binding resource from the broker.